File: chatwoot/python/src/UpdateACustomFilterExample.py
import json
from datetime import date, datetime
from pprint import pprint

from chatwoot_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"userApiKey": "USER_API_KEY"},
    # api_key={"agentBotApiKey": "AGENT_BOT_API_KEY"},
    # api_key={"platformAppApiKey": "PLATFORM_APP_API_KEY"},
)

with ApiClient(configuration) as api_client:
    custom_filter_create_update_payload = models.CustomFilterCreateUpdatePayload(
        name=None,
        type=None,
    )

    try:
        response = api.CustomFiltersApi(api_client).update_a_custom_filter(
            account_id=None,
            custom_filter_id=None,
            custom_filter_create_update_payload=custom_filter_create_update_payload,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling CustomFiltersApi#update_a_custom_filter: %s\n" % e)

File: chatwoot/ruby/src/UpdateACustomFilterExample.rb
require "json"
require "chatwoot_client"

ChatwootClient.configure do |config|
    config.api_key["api_access_token"] = "USER_API_KEY"
    # config.api_key["api_access_token"] = "AGENT_BOT_API_KEY"
    # config.api_key["api_access_token"] = "PLATFORM_APP_API_KEY"
end

custom_filter_create_update_payload = ChatwootClient::CustomFilterCreateUpdatePayload.new

begin
    response = ChatwootClient::CustomFiltersApi.new.update_a_custom_filter(
        0, # account_id
        0, # custom_filter_id
        custom_filter_create_update_payload, # data
    )

    p response
rescue ChatwootClient::ApiError => e
    puts "Exception when calling CustomFiltersApi#update_a_custom_filter: #{e}"
end
